internal/cmd/local/command_kafka_start.go

```diff
@@ -0,0 +1,170 @@
+package local
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"runtime"
+	"strconv"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/api/types/strslice"
+	"github.com/docker/docker/client"
+	"github.com/docker/go-connections/nat"
+	specsv1 "github.com/opencontainers/image-spec/specs-go/v1"
+	"github.com/phayes/freeport"
+	"github.com/spf13/cobra"
+
+	v1 "github.com/confluentinc/cli/internal/pkg/config/v1"
+	"github.com/confluentinc/cli/internal/pkg/errors"
+	"github.com/confluentinc/cli/internal/pkg/log"
+	"github.com/confluentinc/cli/internal/pkg/output"
+)
+
+func (c *command) newKafkaStartCommand() *cobra.Command {
+	return &cobra.Command{
+		Use:   "start",
+		Short: "Start a single-node instance of Apache Kafka.",
+		Args:  cobra.NoArgs,
+		RunE:  c.kafkaStart,
+	}
+}
+
+func (c *command) kafkaStart(cmd *cobra.Command, args []string) error {
+	dockerClient, err := client.NewClientWithOpts(client.FromEnv, client.WithAPIVersionNegotiation())
+	if err != nil {
+		return err
+	}
+	defer dockerClient.Close()
+
+	if err := checkIsDockerRunning(dockerClient); err != nil {
+		return err
+	}
+
+	out, err := dockerClient.ImagePull(context.Background(), confluentLocalImageName, types.ImagePullOptions{})
+	if err != nil {
+		return err
+	}
+	defer out.Close()
+	_, err = io.Copy(os.Stdout, out)
+	if err != nil {
+		return err
+	}
+	log.CliLogger.Tracef("Pull confluent-local image success")
+
+	err = c.prepareAndSaveLocalPorts()
+	if err != nil {
+		return err
+	}
+
+	if c.Config.LocalPorts == nil {
+		return errors.NewErrorWithSuggestions(errors.FailedToReadPortsErrorMsg, errors.FailedToReadPortsSuggestions)
+	}
+	ports := c.Config.LocalPorts
+	platform := &specsv1.Platform{OS: "linux", Architecture: runtime.GOARCH}
+	natKafkaRestPort := nat.Port(ports.KafkaRestPort + "/tcp")
+	natPlaintextPort := nat.Port(ports.PlaintextPort + "/tcp")
+	config := &container.Config{
+		Image:    confluentLocalImageName,
+		Hostname: "broker",
+		Cmd:      strslice.StrSlice{"bash", "-c", "'/etc/confluent/docker/run'"},
+		ExposedPorts: nat.PortSet{
+			natKafkaRestPort: struct{}{},
+			natPlaintextPort: struct{}{},
+		},
+		Env: getContainerEnvironmentWithPorts(ports),
+	}
+	hostConfig := &container.HostConfig{
+		PortBindings: nat.PortMap{
+			natKafkaRestPort: []nat.PortBinding{
+				{
+					HostIP:   localhost,
+					HostPort: ports.KafkaRestPort,
+				},
+			},
+			natPlaintextPort: []nat.PortBinding{
+				{
+					HostIP:   localhost,
+					HostPort: ports.PlaintextPort,
+				},
+			},
+		},
+	}
```

Comment:
There's only one field, so we can remove some indentation here.
```suggestion
	hostConfig := &container.HostConfig{PortBindings: nat.PortMap{
		natKafkaRestPort: []nat.PortBinding{
			{
				HostIP:   localhost,
				HostPort: ports.KafkaRestPort,
			},
		},
		natPlaintextPort: []nat.PortBinding{
			{
				HostIP:   localhost,
				HostPort: ports.PlaintextPort,
			},
		},
	}}
```